src/ui/PrintableUri/PrintableUri.ts

```diff
@@ -39,142 +40,79 @@ export class PrintableUri extends Component {
-
     let tokens: HTMLElement[] = [];
-    let seperators: HTMLElement[] = [];
-
+    let separators: HTMLElement[] = [];
     for (let i = 0; i < parents.length; i++) {
       if (i > 0) {
-        let seperator = this.buildSeperator();
-        seperators.push(seperator);
+        let seperator = this.buildSeparator();
+        separators.push(seperator);
         element.appendChild(seperator);
       }
-
       let parent = <Element>parents.item(i);
       let token = this.buildHtmlToken(parent.getAttribute('name'), parent.getAttribute('uri'));
       tokens.push(token);
-      element.appendChild(token);
-    }
 
-    if (tokens.length > 1) {
-      let ellipsis: HTMLElement = this.buildEllipsis();
-      element.insertBefore(ellipsis, seperators[0]);
-      let ellipsisSeperator: HTMLElement = this.buildSeperator();
-      element.insertBefore(ellipsisSeperator, ellipsis);
-
-      let contentWidth = 0;
-      let tokensWidth: number[] = [];
-      for (let i = 0; i < tokens.length; i++) {
-        tokensWidth[i] = tokens[i].offsetWidth;
-        contentWidth += tokensWidth[i];
-      }
-      let seperatorWidth = seperators[0].offsetWidth;
-      let ellipsisWidth = ellipsis.offsetWidth;
-      let availableWidth = element.offsetWidth;
-
-      if (availableWidth <= contentWidth) {
-        contentWidth += ellipsisWidth + seperatorWidth;
-        let hidden: HTMLElement[] = [];
-        let i = 1;
-        while (i < tokens.length && availableWidth <= contentWidth) {
-          element.removeChild(tokens[i]);
-          element.removeChild(seperators[i - 1]);
-          if (i > 1) {
-            hidden.push(seperators[i - 1]);
-          }
-          hidden.push(tokens[i]);
-          contentWidth -= tokensWidth[i] + seperatorWidth;
-          i++;
-        }
-        ellipsis.onclick = () => {
-          for (let i = 0; i < hidden.length; i++) {
-            element.insertBefore(hidden[i], ellipsis);
-          }
-          element.removeChild(ellipsis);
-        };
-      } else {
-        element.removeChild(ellipsis);
-        element.removeChild(ellipsisSeperator);
-      }
+      element.appendChild(token);
     }
   }
 
   public renderUri(element: HTMLElement, result?: IQueryResult) {
-    this.uri = result.clickUri;
-    let stringAndHoles: StringAndHoles;
-    if (result.printableUri.indexOf('\\') == -1) {
-      stringAndHoles = StringAndHoles.shortenUri(result.printableUri, $$(element).width() / 7);
+    let parentsXml = Utils.getFieldValue(result, 'parents');
+    if (parentsXml) {
+      this.renderParentsXml(element, parentsXml);
     } else {
-      stringAndHoles = StringAndHoles.shortenPath(result.printableUri, $$(element).width() / 7);
+      if (!this.options.titleTemplate) {
+        this.uri = result.clickUri;
+        let stringAndHoles: StringAndHoles;
+        if (result.printableUri.indexOf('\\') == -1) {
+          stringAndHoles = StringAndHoles.shortenUri(result.printableUri, $$(element).width() / 7);
```

Comment:
What's the `/ 7` purpose here ?

I understand it was there before, and you mostly imported the old implementation, but I would like if we had an explanation for this magic number/ a comment that explains why we are dividing by 7 like that.